extensions/binding/org.eclipse.smarthome.binding.openweathermap/src/main/java/org/eclipse/smarthome/binding/openweathermap/internal/discovery/OpenWeatherMapLocationDiscoveryService.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.openweathermap.internal.discovery;
+
+import static org.eclipse.smarthome.binding.openweathermap.OpenWeatherMapBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.binding.openweathermap.handler.AbstractOpenWeatherMapHandler;
+import org.eclipse.smarthome.binding.openweathermap.handler.OpenWeatherMapAPIHandler;
+import org.eclipse.smarthome.binding.openweathermap.internal.config.OpenWeatherMapAPIConfiguration;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.i18n.LocaleProvider;
+import org.eclipse.smarthome.core.i18n.LocationProvider;
+import org.eclipse.smarthome.core.i18n.TranslationProvider;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWeatherMapLocationDiscoveryService} creates things based on the configured location.
+ *
+ * @author Christoph Weitkamp - Initial Contribution
+ */
+@NonNullByDefault
+public class OpenWeatherMapLocationDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWeatherMapLocationDiscoveryService.class);
+
+    private static final int DISCOVERY_TIMEOUT_SECONDS = 30;
+    private static final int DISCOVERY_INTERVAL_SECONDS = 60;
+    private @Nullable ScheduledFuture<?> discoveryJob;
+    private final LocationProvider locationProvider;
+    private @Nullable PointType previousLocation;
+
+    private final OpenWeatherMapAPIHandler bridgeHandler;
+
+    /**
+     * Creates an OpenWeatherMapLocationDiscoveryService.
+     */
+    public OpenWeatherMapLocationDiscoveryService(OpenWeatherMapAPIHandler bridgeHandler,
+            LocationProvider locationProvider, LocaleProvider localeProvider, TranslationProvider i18nProvider) {
+        super(AbstractOpenWeatherMapHandler.SUPPORTED_THING_TYPES, DISCOVERY_TIMEOUT_SECONDS);
+        this.bridgeHandler = bridgeHandler;
+        this.locationProvider = locationProvider;
+        this.localeProvider = localeProvider;
+        this.i18nProvider = i18nProvider;
+        activate(null);
+    }
+
+    @Activate
```

Comment:
Since this service is only created from the handler factory the annotation is not necessary.